TaylorFramework/Modules/caprices/ProccessingMessage/OptionsProcessor.swift

```diff
@@ -45,10 +45,10 @@ class OptionsProcessor {
         return resultDictionary
     }
     
-    func processArguments(arguments: [String]) throws -> [Option] {
-         return try Array(arguments[1..<arguments.count]).reduceTwoElements([]) {
+    func processArguments(_ arguments: [String]) throws -> [Option] {
+         return try Array(arguments[1..<arguments.count]).reduceTwoElements(initial: []) {
```

Comment:
`Array(arguments[1..<arguments.count])` is the same as `arguments.dropFirst()`. Just change `reduceTwoElements` to be extension on `Sequence`:
```
extension Sequence {
    func reduceTwoElements<T>(_ initial: T, combine: (T, Iterator.Element, Iterator.Element) throws -> T) rethrows -> T {
        var result = initial
        for (first, second) in zip(self.enumerated().filter { $0.0.isEven }.map { $0.1 },
                                            self.enumerated().filter {$0.0.isOdd }.map {$0.1}) {
            result = try combine(result, first, second)
        }
        return result
    }
}
```